knative-operator/pkg/controller/knativekafka/knativekafka_controller.go

```diff
@@ -78,7 +86,130 @@ func (r *ReconcileKnativeKafka) Reconcile(request reconcile.Request) (reconcile.
 		return reconcile.Result{}, err
 	}
 
-	// NOT IMPLEMENTED YET
+	// TODO: check for deletion
+	// if original.GetDeletionTimestamp() != nil {
+	//		return reconcile.Result{}, r.delete(original)
+	//	}
+
+	instance := original.DeepCopy()
+	reconcileErr := r.reconcileKnativeKafka(instance)
+
+	if !equality.Semantic.DeepEqual(original.Status, instance.Status) {
+		if err := r.client.Status().Update(context.TODO(), instance); err != nil {
+			return reconcile.Result{}, fmt.Errorf("failed to update status: %w", err)
+		}
+	}
+
+	if instance.Status.IsReady() {
+		common.KnativeKafkaUpG.Set(1)
+	} else {
+		common.KnativeKafkaUpG.Set(0)
+	}
+	return reconcile.Result{}, reconcileErr
+}
+
+func (r *ReconcileKnativeKafka) reconcileKnativeKafka(instance *operatorv1alpha1.KnativeKafka) error {
+	stages := []func(*operatorv1alpha1.KnativeKafka) error{
+		// TODO r.configure,
+		// TODO r.ensureFinalizers,
+		r.installKnativeKafka,
+	}
+	for _, stage := range stages {
+		if err := stage(instance); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Install Knative Kafka components
+func (r *ReconcileKnativeKafka) installKnativeKafka(instance *operatorv1alpha1.KnativeKafka) error {
+	if err := applyKnativeKafka(instance, r.client); err != nil {
+		instance.Status.MarkDependencyInstalling("KnativeKafka")
```

Comment:
This is not a dependency imo. We should probably use the `InstallSucceeded` condition for this and completely drop the dependency condition.

Reply:
+1